File: 891-score-after-flipping-matrix/score-after-flipping-matrix.cpp
class Solution {
    void flipr(vector<vector<int>>&grid,int i){
        for(int j=0;j<grid[0].size();j++){
            grid[i][j]=!grid[i][j];
        }
    }
    void flipc(vector<vector<int>>&grid,int j){
        for(int i=0;i<grid.size();i++){
            grid[i][j]=(grid[i][j]==0?1:0);
        }
    }

public:

    int matrixScore(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        for(int i=0;i<m;i++){
            if(grid[i][0]==0){
                flipr(grid,i);
            }
        }
        for(int j=1;j<n;j++){
            int zeroes = 0;
            int ones = 0;
            for(int i=0;i<m;i++){
                zeroes+=(grid[i][j]==0?1:0);
                ones+=grid[i][j];
            }
            if(zeroes>ones){
            // cout<<zeroes<<" "<<ones<<endl;
                
                flipc(grid,j);
            }
        }
        int p=0;
        for(int i=0;i<m;i++){
            int num = 0;
            for(int j = n-1;j>=0;j--){
                if(grid[i][j]==1){

                num+=pow(2,n-1-j);
                }
            

            }
            // cout<<num<<endl;
            p+=num;
        }
        
        return p;

    }
};

File: 891-score-after-flipping-matrix/score-after-flipping-matrix.py
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        def flipr(self,grid: List[List[int]],i):
            for j in range(len(grid[0])):
                grid[i][j] = 0 if grid[i][j] == 1 else 1
        def flipc(self,grid: List[List[int]],j):
            for i in range(len(grid)):
                grid[i][j] = 0 if grid[i][j] == 1 else 1
        m = len(grid)
        n = len(grid[0])
        for i in range(m):
            if grid[i][0]==0:
                flipr(self,grid,i)
        
        for j in range(1,n):
            zeroes = 0
            ones = 0
            for i in range(m):
                if(grid[i][j]==0):
                    zeroes+=1
                
                else:
                    ones+=1
            
            if(zeroes>ones):
                flipc(self,grid,j)
        ans = 0
        for i in range(m):
            for j in range(n-1,-1,-1):
                if grid[i][j]==1:
                    ans+=pow(2,n-1-j)

        return ans
